composer/trainer/trainer.py

```diff
@@ -898,7 +906,56 @@ def __init__(
 
         # Load Checkpoint
         self._rng_state = None
-        if load_path is not None:
+        # If autoresume is enabled, first check for existing checkpoints to load
```

Comment:
can this be moved into a separate helper method so that the trainer code is more readable? e.g. ideally something like:

```
if autoresume:
    latest_checkpoint = retrieve_latest_checkpoint(...)
    if latest_checkpoint:
        log.info('Found XYZ, ...')
        load_path = latest_checkpoint

load_checkpoint(...)
```
or maybe just putting `autoresume` into the `load_checkpoint` function?
```
load_checkpoint(..., autoresume=autoresume)
```

Prefer not to have two places where we call load_checkpoint.

Reply:
Picked the first option since we need to ignore certain args (object_store, weights_only) with autoresume

Reply:
Ah to clarify, could you move all the auto-resume specific logic (the try except around the logger, etc.) out into a separate helper method (and out of the trainer logic)

Reply:
Oh yes, sorry I know understand the code snippet. Updating!